itou/www/search_views/views.py

```diff
@@ -50,6 +62,9 @@ def get(self, request, *args, **kwargs):
         return self.post(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
+        saved_searches = list(self.request.user.saved_searches.all()) if self.request.user.is_authenticated else []
```

Comment:
Je ferais plutôt le cast à un seul endroit dans `SavedSearch.add_city_name_attr` car il est facile d'oublier : https://github.com/gip-inclusion/les-emplois/pull/6757/files#diff-b606b7c5ce6a7bb9b226c4ace363625e83aebd739d1658e5a56cb78acd904846R369

Idéalement pour _éviter_ le polymorphisme, il faudrait ici :

```suggestion
        saved_searches = self.request.user.saved_searches.all() if self.request.user.is_authenticated else SavedSearch.objects.none()
```

Puis la signature de `SavedSearch.add_city_name_attr` serait la suivante :

```python
    def add_city_name_attr(saved_searches: models.QuerySet) -> list[SavedSearch]:
        saved_searches = list(saved_searches)
        # …
        return saved_searches
```

Qu'on soit d'accord, ça change pas grand chose fondamentalement, mais je préfère passer une liste au contexte qu'un queryset évalué, dont les objets cachés ont été annotés en python.